pkg/integrations/quay.go

```diff
@@ -0,0 +1,182 @@
+package integrations
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"net/url"
+	"strings"
+
+	"github.com/redhat-appstudio/rhtap-cli/pkg/config"
+	"github.com/redhat-appstudio/rhtap-cli/pkg/k8s"
+
+	"github.com/spf13/pflag"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// QuayIntegration represents the RHTAP Quay integration.
+type QuayIntegration struct {
+	logger *slog.Logger   // application logger
+	cfg    *config.Config // installer configuration
+	kube   *k8s.Kube      // kubernetes client
+
+	force bool // overwrite the existing secret
+
+	dockerconfigjson string // dockerconfig credentials
+	token            string // API token credentials
+	url              string // quay URL
+}
+
+// PersistentFlags sets the persistent flags for the Quay integration.
+func (q *QuayIntegration) PersistentFlags(p *pflag.FlagSet) {
+	p.BoolVar(&q.force, "force", q.force,
+		"Overwrite the existing secret")
+
+	p.StringVar(&q.dockerconfigjson, "dockerconfigjson", q.dockerconfigjson,
+		"Quay dockerconfigjson")
+	p.StringVar(&q.token, "token", q.token,
+		"Quay API token")
+	p.StringVar(&q.url, "url", q.url,
+		"Quay URL")
+}
+
+// log logger with contextual information.
+func (q *QuayIntegration) log() *slog.Logger {
+	return q.logger.With(
+		"url", q.url,
+		"force", q.force,
+		"dockerconfigjson-len", len(q.dockerconfigjson),
+		"token-len", len(q.token),
+	)
+}
+
+// Validate checks if the required configuration is set.
+func (q *QuayIntegration) Validate() error {
+	if q.dockerconfigjson == "" {
+		return fmt.Errorf("dockerconfigjson is required")
+	}
+	if q.token == "" {
+		return fmt.Errorf("token is required")
+	}
+	if q.url == "" {
+		return fmt.Errorf("url is required")
+	} else {
+		u, err := url.Parse(q.url)
+		if err != nil {
+			return fmt.Errorf("invalid url")
+		}
+		if !strings.HasPrefix(u.Scheme, "http") {
+			return fmt.Errorf("invalid url scheme, expected one of 'http', 'https'")
+		}
+	}
+	return nil
+}
+
+// EnsureNamespace ensures the namespace needed for the Quay integration secret
+// is created on the cluster.
+func (q *QuayIntegration) EnsureNamespace(ctx context.Context) error {
+	feature, err := q.cfg.GetFeature(config.RedHatDeveloperHub)
+	if err != nil {
+		return err
+	}
+	return k8s.EnsureOpenShiftProject(
+		ctx,
+		q.log(),
+		q.kube,
+		feature.GetNamespace(),
+	)
+}
+
+// secretName returns the secret name for the integration. The name is "lazy"
+// generated to make sure configuration is already loaded.
+func (q *QuayIntegration) secretName() types.NamespacedName {
+	feature, _ := q.cfg.GetFeature(config.RedHatDeveloperHub)
+	return types.NamespacedName{
+		Namespace: feature.GetNamespace(),
+		Name:      "rhtap-quay-integration",
+	}
+}
```

Comment:
For the record, later on we should export this type of routine to a common place. There's a couple examples this and ensure the installer namespace is created, etc.

Reply:
Agreed.